client.go

```diff
@@ -163,25 +167,35 @@ func (c Client) findIDInConfigurationsPayload(body io.ReadCloser, name string) (
 	return "", err
 }
 
+// ListNodes returns a list of dataset agent nodes from Flocker Control Service
+func (c *Client) ListNodes() (nodes []NodeState, err error) {
+	resp, err := c.get(c.getURL("state/nodes"))
+	if err != nil {
+		return
```

Comment:
Would typically like to see an actual error returned. 

The below returns no info
```
if err != nil {
		return
}
```

Preferred something like the below
```
if err != nil {
		return nil , err
}
```

or

```
if err != nil {
		return "" , err
}
```